packages/angular/src/generators/component/component.ts

```diff
@@ -0,0 +1,123 @@
+import { joinPathFragments, logger, names, readJson, Tree } from '@nrwl/devkit';
+import type { Schema } from './schema';
+import { wrapAngularDevkitSchematic } from '@nrwl/devkit/ngcli-adapter';
+import {
+  formatFiles,
+  readWorkspaceConfiguration,
+  readProjectConfiguration,
+  normalizePath,
+} from '@nrwl/devkit';
+import { normalize } from 'path';
+
+export async function componentGenerator(tree: Tree, schema: Schema) {
+  checkPathUnderProjectRoot(tree, schema);
+
+  const angularComponentSchematic = wrapAngularDevkitSchematic(
+    '@schematics/angular',
+    'component'
+  );
+  await angularComponentSchematic(tree, {
+    ...schema,
+    skipImport: true,
+    export: false,
+  });
+
+  exportComponent(tree, schema);
+
+  await formatFiles(tree);
+}
+
+function checkPathUnderProjectRoot(tree: Tree, schema: Partial<Schema>) {
+  if (!schema.path) {
+    return;
+  }
+
+  const project =
+    schema.project ?? readWorkspaceConfiguration(tree).defaultProject;
+  const { root } = readProjectConfiguration(tree, project);
+
+  let pathToComponent = normalizePath(schema.path);
+  pathToComponent = pathToComponent.startsWith('/')
+    ? pathToComponent.slice(1)
+    : pathToComponent;
+
+  if (!pathToComponent.startsWith(normalize(root))) {
```

Comment:
We're mixing usages of `normalizePath` and `normalize` here. The former will always produce paths with slashes while the latter will produce backslashes on Windows. This can cause bugs where valid paths won't match the root on Windows. To make comparisons we should use a consistent way of normalizing the paths.